src/Main.elm

```diff
@@ -1,33 +1,77 @@
 module Main exposing (..)
 
 import Browser
-import Html exposing (Html, text, div, h1, img)
-import Html.Attributes exposing (src)
+import Html exposing (..)
+import Html.Attributes exposing (class, href)
+import Html.Events exposing (..)
+import Http
+import Json.Decode as Decode exposing (Decoder, andThen, field, int, list, string, succeed)
+import Json.Decode.Pipeline exposing (required)
+import RemoteData exposing (RemoteData, WebData)
+
+
 
 
 ---- MODEL ----
 
 
+type alias StoryId =
+  Int
+
+
 type alias Model =
-    {}
+    { stories : WebData (List StoryId)
+    }
+
+
+emptyModel : Model
+emptyModel =
+    { stories = RemoteData.Loading
+    }
 
 
 init : ( Model, Cmd Msg )
 init =
-    ( {}, Cmd.none )
+    ( emptyModel, Cmd.none )
 
 
 
 ---- UPDATE ----
 
 
 type Msg
-    = NoOp
+    = HandleFetchedStoryIds (WebData (List StoryId))
+
+
+getStoryApi : Int -> Cmd Msg
+getStoryApi id =
+    Http.get
+        { url = "https://hacker-news.firebaseio.com/v0/item/" ++ ( String.fromInt id ) ++ ".json"
+
+        , expect =
+            list int
+                |> Http.expectJson (RemoteData.fromResult >> HandleFetchedStoryIds )
+        }
+
+
+itemId : Cmd Msg
+itemId =
+    Http.get
+        { url = "https://hacker-news.firebaseio.com/v0/topstories.json"
+        , expect =
+            list int
+                |> Http.expectJson (RemoteData.fromResult >> HandleFetchedStoryIds)
+        }
 
 
 update : Msg -> Model -> ( Model, Cmd Msg )
 update msg model =
-    ( model, Cmd.none )
+    case msg of
+        HandleFetchedStoryIds news ->
```

Comment:
@Diama1 
Note that here we have the IDs, so we can use them to create new API requests.
We don't need to store them.